src/main/java/com/softserve/academy/spaced/repetition/service/RegistrationService.java

```diff
@@ -20,20 +21,33 @@
 @Service
 public class RegistrationService {
 
+    private final AuthorityRepository authorityRepository;
+    private final UserService userService;
+    private final PasswordEncoder passwordEncoder;
+    private final MailService mailService;
+    private final EmailUniquesValidator emailUniquesValidator;
+    private final BlankFieldValidator blankFieldValidator;
+    private final NullFieldsValidator nullFieldsValidator;
+    private final RememberingLevelRepository rememberingLevelRepository;
+
     @Autowired
-    private AuthorityRepository authorityRepository;
-    @Autowired
-    private UserService userService;
-    @Autowired
-    private PasswordEncoder passwordEncoder;
-    @Autowired
-    private MailService mailService;
-    @Autowired
-    private EmailUniquesValidator emailUniquesValidator;
-    @Autowired
-    private BlankFieldValidator blankFieldValidator;
-    @Autowired
-    private NullFieldsValidator nullFieldsValidator;
+    public RegistrationService(AuthorityRepository authorityRepository,
```

Comment:
Sonar will report an issue.

Reply:
Well, what can I do? I suppose all the parameters are needed. At least, I have added only the latest. It's not good to autowire fields not in the constructor so I refactored that.

Reply:
Except constructor and field autowiring, you can use setters.

Reply:
Well, I'll try, whether it'll work. Thanks.